File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.py
class Solution(object):
    def minOperations(self, li, x):
        numsSum = sum(li)
        k = numsSum-x
        n = len(li)
        i, j, s = 0, 0, 0
        l = []
        maximum = float('-inf')
        if numsSum == x:
            return n
        if k>0:
            while j < n:
                s += li[j]
                if s < k:
                    j += 1
                elif s == k:
                    maximum = max(maximum, j-i+1)
                    j += 1
                elif s > k:
                    while s > k:
                        s -= li[i]
                        i += 1
                        if s == k:
                            maximum = max(maximum, j-i+1)
                    j += 1
            return n-maximum if maximum != float('-inf') else -1
        else: return -1

File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.java
class Solution {
    public int minOperations(int[] nums, int x) {
        int sum = 0;
        for(int i =0;i<nums.length;i++){
            sum=sum+nums[i];
        }
        int target = sum - x;
        if(target<0){
            return -1;
        }
        if(target==0){
            return nums.length;
        }

        int i=0;
        int j=0;
        int temp =0;
        int max = 0;
        while(j<nums.length){
            temp = temp+nums[j];
            if(temp<target){
                j++;
            }
            else if (temp==target){
                max = Math.max(max,j-i+1);
                j++;
            }
            else{
                while(temp>target){
                    temp = temp - nums[i];
                    i++;
                }
                if(temp==target){
                    max = Math.max(max,j-i+1);
                }
                j++;
            }
        }
        if(max==0){
            return -1;
        }
        return nums.length-max;
    }
}